### ISSUE TITLE
msal redirect to the app using B2C not working, but login screen opened in a new browser when localhost is set.

### ISSUE THREAD

**Issue by satyajit-behera** [Azure-Samples/ms-identity-dotnetcore-maui#18]
When I use the sample with required changes, it gives me the error:
Only loopback redirect uri is supported, but msalId://auth/ was found. Configure http://localhost or http://localhost:port both during app registration and when you create the PublicClientApplication object. See https://aka.ms/msal-net-os-browser for details

But when i set the loopback url as localhost, it opens up in a new browser window and loopback happens from there. Can we make it open within the same application windows in an MAUI application.

Thanks

**Comment by DRAirey1** [comment]
Yes. Ran across the same thing with a WinUI application. I came here looking for a working example, only to find out that this example is broken as well. When I replaced the normal callback with the 'localhost' version, I got a system browser instead of the normal browser. That part was bad, but what is worse is the user flows are all messed up. I can't seem to create a new user, or link to external providers.

**Comment by Conbag93** [comment]
Can someone at Microsoft at least acknowledge this issue?

I'm trying to do everything by the book here - using the Microsoft recommended UI framework (MAUI) with the Microsoft recommended go-to when it comes to authenticating external users (Azure AD B2C).

What @DRAirey1 said is correct, and it's honestly unacceptably poor for any Enterprise application.

**Comment by ncds42** [comment]
I was having issues with getting this sample to run, and found that the way the redirect URI was being built was the source of the issue. When you create the Application Registration in B2C it will generate the redirect URI for you. This does not get replaced by the appsettings.json file properly, so overriding it manually in PlatformConfig.cs, using the value copied from B2C Tenant Application Registration, solved the issue for me.
